jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -286,6 +286,29 @@ public static void putIntLittleEndian(ByteBuffer buffer, int value)
         flipToFlush(buffer, p);
     }
 
+    /**
+     * Slice a buffer given an offset and a length.
+     * @param buffer the buffer to slice
+     * @param offset the offset
+     * @param length the length, -1 meaning use the current limit
+     * @return the sliced buffer
+     */
+    public static ByteBuffer slice(ByteBuffer buffer, int offset, int length)
+    {
+        ByteBuffer slice = buffer.slice();
+        if (offset > 0)
```

Comment:
This doesn't check offset < 0, which should throw an `IndexOutOfBoundsException` (as `ByteBuffer#slice(int,int)`) (or `IllegalArgumentException`)

Reply:
Added `< 0` check.